lib/inertia_rails/renderer.rb

```diff
@@ -166,5 +146,72 @@ def excluded_by_only_partial_keys?(path_with_prefixes)
     def excluded_by_except_partial_keys?(path_with_prefixes)
       partial_except_keys.present? && (path_with_prefixes & partial_except_keys).any?
     end
+
+    def validate_partial_reload_optimization
+      if prop_transformer.unoptimized_prop_paths.any?
+        case configuration.action_on_unoptimized_partial_reload
+        when :log
+          ActiveSupport::Notifications.instrument(
+            'inertia_rails.unoptimized_partial_render',
+            paths: prop_transformer.unoptimized_prop_paths,
+            controller: controller,
+            action: controller.action_name,
```

Comment:
@skryukov you mentioned wanting this feature to accept a callable... with this anyone can register a subscriber with that sort of functionality. any other options you'd like to see passed to the subscriber context?